File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer = {arr[0]};
    
    for (int i=1; i<arr.size(); i++) {
        if (arr[i] != arr[i-1]) {
            answer.push_back(arr[i]);
        }   
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> intList = new ArrayList<>();
        int pre = -1;
        for (int a : arr) {
            if (a != pre) {
                intList.add(a);
            }
            pre = a;
        }
        
        int[] answer = new int[intList.size()];
        for (int i=0; i<intList.size(); i++) {
            answer[i] = intList.get(i);
        }
        
        return answer;
    }
}